### DISCUSSION TITLE
Trailing Checksums for S3 Streaming Uploads

### DISCUSSION THREAD

**Discussion by normelton** [aws/aws-sdk-ruby#2768] [Q&A]
The AWS S3 documentation implies that I can specify a ChecksumAlgorithm when starting a streaming upload, and that the SDK will automagically calculate the checksum of uploaded data, and provide it to S3 at the completion of the upload:

https://docs.aws.amazon.com/AmazonS3/latest/userguide/checking-object-integrity.html#trailing-checksums

And in fact I see references to this in the `gems/aws-sdk-s3/CHANGELOG.md` file, but am having trouble getting it to work. I tried:

```
object.upload_stream({checksum_algorithm: "CRC32"}) do |s3_stream|
```

But I get back: **Aws::S3::Errors::InvalidRequest The upload was created using a crc32 checksum. The complete request must include the checksum for each part. It was missing for part 1 in the request.**

Any suggestions?

**Comment by alextwoods** [comment]
Sorry you're running into this - this looks like it may be a bug - can you open an issue and I'll start investigating.

**Reply by normelton** [reply]
Thanks for the quick reply, opened #2769. Let me know if I can help test anything!
